### DISCUSSION TITLE
Has anybody been able to use the new V3 Api

### DISCUSSION THREAD

**Discussion by bilpor** [etsy/open-api#625] [Q&A] [answered]
Hi All, I was using V2 of the Api, but a certain feature wasn't working and the Etsy Developers confirmed it as a bug and suggested I go to V3. I've converter to try and use V3 but I can't get past 2 issues. The first is that getting receipts by shop id was fine in V2 as the shopId was a string usually your shop name. In V3 the changed it to an integer and when creating your shop there's no way of setting or seeing this. I was then told to use getShopByOwnerUserId again the same issue, userId is an integer which I cant set  or see anywhere.

After several emails to support I asked them if they have actually created a shop put a few fake orders in and then tested their Api and if so could they send the sequence of calls they have made to obtain any data. THey could not provide me with this information but instead directed me here. Can anybody give an example of the sequence of calls that need to be made to get a shop id or even find out what that id is. At the moment we are completely blocked.

**Comment by Gareth064** [comment]
So in Etsy, A shop has one owner.
When you get an Access Token, the first part of the token, before the period, is the UserId\OwnerId of the Shop that the Access Token is for.
![image](https://user-images.githubusercontent.com/22307223/195808810-1e1290a4-7110-416d-b7d5-2247c2a6de63.png)

Extract that from the Access Token, and now you can use the `getShopByOwnerUserId` endpoint

**Reply by bilpor** [reply]
Thank you Gareth, I will give that a try, hopefully this is the Achiles heel that I need to get me going again.

**Comment by etsyachristensen** [comment, marked as answer]
We have just implemented a new endpoint that you can call after a user authorizes your app:

https://developers.etsy.com/documentation/reference#operation/getMe

This will return the shop_id and user_id for the given authenticated user.

**Comment by bilpor** [comment]
Just tried this, and it never returns from the call. I have raised a bug.

**Reply by etsyachristensen** [reply]
@bilpor Can you email the response headers from the api call? Or are you not even getting any response at all? Are you providing the bearer token in the call? Is this happening for every user or just a particular user?

**Comment by bilpor** [comment]
I do not receive any response at all. If I go back to starting with the user id from the authentication, then every call I make after that works. For this particular call, I set all tokens etc in the header in the same way I set up all other get calls. 

 

I can’t provide a curl call as I’m not using that. I am using C# I provided my call method which sets up the header before passing. We only have 1 user set up.

 

Kind Regards

Will

**Reply by coder771** [reply]
@bilpor did it ever work for you. can you please put a sample of your request. we are having similar issues

**Reply by bilpor** [reply]
No, I could not get the details from the GetMe endpoint. The call never returned. I split the values from the authentication response and used that instead.

**Reply by coder771** [reply]
thanks for getting back. I'm using exchange token for now as we have missed deadline, so just trying to get it done as quick for existing users.

**Comment by alihammadraza** [comment]
visit etsy.com/shop/your_shops_name and copy the url of Contact button next to the envelope icon on the right. In this url there are multiple ids, one of them is the shop id you need for the api. It was the referring id in my case.
